src/Utilities/RunTimeManager.cpp

```diff
@@ -129,6 +130,21 @@ bool RunTimeControl<CLOCK>::checkStop(LoopTimer<CLOCK>& loop_timer)
   return need_to_stop;
 }
 
+template<class CLOCK>
+std::string RunTimeControl<CLOCK>::generateProgressMessage(const std::string& driverName,
+                                                           int block,
+                                                           int num_blocks) const
+{
+  std::stringstream log;
+  if (block == 0 || block == num_blocks / 4 || block == num_blocks / 2 || block == (num_blocks * 3) / 4 ||
+      block + 1 == num_blocks)
+  {
+    log << "Completed block " << std::setw(4) << block + 1 << " of " << num_blocks << " average "
+        << std::setprecision(4) << m_loop_time << " secs/block after " << m_elapsed << " secs" << std::endl;
```

Comment:
You will need another `std::setprecision(4)`  before `m_elapsed` since the previous got cleared by a string printout.

Reply:
Done.